src/client/fontengine.cpp

```diff
@@ -280,8 +284,17 @@ gui::IGUIFont *FontEngine::initFont(const FontSpec &spec)
 	if (spec.italic)
 		setting_suffix.append("_italic");
 
-	u32 size = std::floor(RenderingEngine::getDisplayDensity() *
-			g_settings->getFloat("gui_scaling") * spec.size);
+	// Apply GUI scaling to the font size
+	u32 size = spec.size * g_settings->getFloat("gui_scaling");
+
+	// Constrain the font size to a certain multiple, if necessary
+	u16 multiple_of = 1;
+	g_settings->getU16NoEx(setting_prefix + "font_require_multiple", multiple_of);
+	if (multiple_of != 1)
+		size = (u32)std::round((double)size / multiple_of) * multiple_of;
+
+	// Convert size from points to pixels (a point is equal to a pixel at 72 DPI)
+	size *= RenderingEngine::getDisplayDensity() * 96 / 72;
```

Comment:
does this work correctly when formspec code already tries to scale everything by display density (or something)?

Reply:
Formspecs only scale coordinate sizes, so FontEngine scales the fonts itself for everything (both formspecs and HUD)